app/shared/components/footer/footer.component.ts

```diff
@@ -1,32 +1,44 @@
-import {Component, OnInit} from 'angular2/core';
-import {RouterLink} from 'angular2/router';
-
-import {Observable} from 'rxjs/Observable';
+import {Component, OnInit} from '@angular/core';
+import {RouterLink} from '@angular/router-deprecated';
 import {ContenfulContent} from '../../services/contentful-content.service';
-import {AsyncPipe} from 'angular2/common';
-import {ContentfulConfig} from '../../../app.constans';
-import {ContentfulPageStructure} from '../../structures/aliases.structures';
+import {AsyncPipe} from '@angular/common';
+import {Menu} from '../../structures/content-type.structures';
+import {RoutesGatewayService} from '../../services/routes-gateway.service';
+import {Angulartics2On} from 'angulartics2/index';
 
 @Component({
-  selector: 'footer',
-  template: <string> require('./footer.html'),
-  directives: [RouterLink],
-  styles: [<string> require('./footer.styl')],
+  selector: 'gm-footer',
+  template: require('./footer.html') as string,
+  directives: [RouterLink, Angulartics2On],
+  styles: [require('./footer.styl') as string],
   pipes: [AsyncPipe]
-
 })
-
-export class Footer implements OnInit {
-  private children: Observable<ContentfulPageStructure[]>;
-
-  constructor(private _contentfulContent: ContenfulContent) {
+export class FooterComponent implements OnInit {
+  private typeMenu: string = 'footerMenu';
+  private menu: Menu[];
+  private slug: string;
+  private contentfulContent: ContenfulContent;
+  private routesGatewayService: RoutesGatewayService;
+
+  public constructor(contentfulContent: ContenfulContent,
+                     routesGatewayService: RoutesGatewayService) {
+    this.contentfulContent = contentfulContent;
+    this.routesGatewayService = routesGatewayService;
   }
 
-  ngOnInit(): void {
-
-    this.children = this._contentfulContent
-      .getPageTree(ContentfulConfig.CONTENTFUL_PAGE_TREE_FOOTER_ID)
-      .map(response => response.structure.children);
-
+  public ngOnInit(): void {
+    this.contentfulContent
+      .getMenu(this.typeMenu)
+      .subscribe((response: any) => {
+        this.menu = response[0].fields.entries;
+        for (let item of this.menu) {
+          if (item.fields.entryPoint) {
+            let title = item.fields.entryPoint.fields.title;
+
+            this.slug = item.fields.entryPoint.fields.slug;
```

Comment:
same as above - if slug is not used in a template or by other comps - then use local vars (let)